assets/dev/js/editor/utils/jquery-html5-dnd.js

```diff
@@ -188,9 +188,15 @@
 				return;
 			}
 
-			var insertMethod = 'top' === currentSide ? 'prependTo' : 'appendTo';
-
-			elementsCache.$placeholder[ insertMethod ]( currentElement );
+			// Fix placeholder placement for Container with `flex-direction: row`.
+			if ( ! $( currentElement ).hasClass( 'elementor-first-add' ) && $( currentElement ).parents( '.e-container--placeholder-row' ).length ) {
```

Comment:
Work with the UI state

Reply:
Seems like I can't, since the direction state is available only when editing an element.